ams/server/src/test/java/com/netease/arctic/server/metrics/TestMetricsEmitterLoader.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netease.arctic.server.metrics;
+
+import com.netease.arctic.ams.api.metrics.MetricType;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Objects;
+
+import static com.netease.arctic.ams.api.Environments.AMORO_HOME;
+
+public class TestMetricsEmitterLoader {
```

Comment:
Why do we need a Loader class instead of loading in MetricsMananger?
Is this really necessary?

Reply:
This class may not be necessary, but there is indeed a dilemma here regarding how to obtain the pluginName. The current implementation reads the configuration file, but it is clear that the responsibility of reading the configuration file should be handled by com.netease.arctic.server.manager.MetricsManager#loadProperties, and external objects should not know the format of the configuration file, and therefore do not know the pluginName. It would be better to extract the responsibility of reading the configuration file from the PluginManager and assign it to a separate class.